controllers/progressivesync_controller.go

```diff
@@ -99,11 +108,26 @@ func (r *ProgressiveSyncReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	pss, _ := r.StateManager.Get(ps.Name)
+
+	newHash, err := r.calculateHashedSpec(ctx, &ps)
+	if err != nil {
+		log.Error(err, "Failed to generate the hash value from the ApplicationSet spec")
+		return ctrl.Result{}, err
+	} else {
+		currentHash := pss.GetHashedSpec()
+		if currentHash != newHash {
+			pss.SetHashedSpec(newHash)
+			log.Info("Successfully updated the hash value of the ApplicationSet spec", "service", ps.Name)
+		} else {
+			log.Info("Hash value is the same as nothing has changed in the ApplicationSet spec", "service", ps.Name)
+		}
+	}
```

Comment:
I think we should be able to simplify this `if/else`
```suggestion
	if err != nil {
		log.Error(err, "Failed to generate the hash value from the ApplicationSet spec")
		return ctrl.Result{}, err
	}
	currentHash := pss.GetHashedSpec()
	if currentHash != newHash {
		pss.SetHashedSpec(newHash)
		log.Info("Successfully updated the hash value of the ApplicationSet spec", "service", ps.Name)
	} else {
		log.Info("Hash value is the same as nothing has changed in the ApplicationSet spec", "service", ps.Name)
	}
```